src/common/datasource/src/file_format.rs

```diff
@@ -124,3 +133,15 @@ pub fn open_with_decoder<T: ArrowDecoder, F: Fn() -> DataFusionResult<T>>(
         Ok(stream.boxed())
     }))
 }
+
+pub async fn infer_schemas(
+    store: &ObjectStore,
+    files: &[String],
+    file_format: impl FileFormat,
+) -> Result<Schema> {
```

Comment:
Changing the return type to ArrowSchema is more clear.
```suggestion
) -> Result<ArrowSchema> {
```